low-level/files/opt/seagate/sspl/setup/sspl_post_install.py

```diff
@@ -207,15 +207,43 @@ def create_sspl_conf(self):
             shutil.copyfile("%s/conf/sspl.conf.%s.yaml" % (consts.SSPL_BASE_DIR,
                 self.product), consts.file_store_config_path)
 
+    def validate_user_created(self):
+        max_retry = 3
+        for i in range(max_retry):
+            sspl_uid = Utility.get_uid(consts.USER)
+            if sspl_uid != -1:
+                break
+            else:
+                if i == (max_retry-1):
+                    msg = "No user found with name : %s" % (consts.USER)
+                    logger.error(msg)
```

Comment:
so no retries to create user again, but just validation again & again. Who would create it then if it fails first time ?

Reply:
I had the same doubt, @sandeepsgt clarified that the failure occurred in validation rather than user creation.

Reply:
Correct @jjxsg . @cdeshmukh if you see logs in [EOS-25084](https://jts.seagate.com/browse/EOS-25084). User was created as [this ](https://github.com/Seagate/cortx-monitor/pull/528/files#diff-e39efed6e2afb5011c788a62bb7f25d9c005e23bc3591fbd201fa42963897903L216) error was not seen, instead error was from [validation](https://github.com/Seagate/cortx-monitor/pull/528/files#diff-e39efed6e2afb5011c788a62bb7f25d9c005e23bc3591fbd201fa42963897903L241)

Reply:
If user creation fails, then we dont have retries for same. We should retry for user creation as well as validation too
and rename function

validate_user_existence()

Reply:
Updated function name

As discussed not adding retries for user creation, as it will not help as user creation can't fail for random reason